### ISSUE TITLE
💅 `useOptionalChain` flagging condition for function calls

### ISSUE THREAD

**Issue by hyoretsu** [biomejs/biome#828]
### Environment information

```bash
CLI:
  Version:                      1.3.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v18.15.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.10.2"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

lint/complexity/useOptionalChain

### Playground link

Too intricate to provide a repro, but easy enough to pinpoint

### Expected result

`command.autocomplete && command.autocomplete(interaction)` isn't a valid target to use optional chaining.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
